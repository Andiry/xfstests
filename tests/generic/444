#! /bin/bash
# FS QA Test 444
#
# Check if SGID is inherited when creating a subdirectory when the owner is not
# in the owning group and directory has default ACLs.
#
#-----------------------------------------------------------------------
# Copyright (c) 2017 SUSE Linux.  All Rights Reserved.
#
# Author: Jan Kara <jack@suse.cz>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License as
# published by the Free Software Foundation.
#
# This program is distributed in the hope that it would be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write the Free Software Foundation,
# Inc.,  51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
#-----------------------------------------------------------------------
#

seq=`basename $0`
seqres=$RESULT_DIR/$seq
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1	# failure is the default!
trap "_cleanup; exit \$status" 0 1 2 3 15

_cleanup()
{
	cd /
	rm -f $tmp.*
}

# get standard environment, filters and checks
. ./common/rc
. ./common/filter
. ./common/attr

# real QA test starts here
_supported_fs generic
_supported_os Linux
_require_test
_require_runas
_require_acls

cd $TEST_DIR

TDIR=testdir.$seq

# Create SGID directory with default ACLs
rm -rf $TDIR
mkdir $TDIR
setfacl -d -m u:101:rwx $TDIR
chown 100:100 $TDIR
chmod 2755 $TDIR

_runas -u 100 -g 100 -- mkdir $TDIR/testsub1
stat -c %A $TDIR/testsub1
_runas -u 100 -g 101 -- mkdir $TDIR/testsub2
stat -c %A $TDIR/testsub2

status=0
exit
