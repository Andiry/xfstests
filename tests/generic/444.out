QA output created by 444
drwxrwsr-x
drwxrwsr-x
